src/gausskernel/storage/access/k2/pg_gate_api.cpp

```diff
@@ -317,8 +387,59 @@ K2PgStatus PgGate_ExecCreateTable(const char *database_name,
                               bool if_not_exist,
                               bool add_primary_key,
                               const std::vector<K2PGColumnDef>& columns) {
-  elog(DEBUG5, "PgGateAPI: PgGate_NewCreateTable %s, %s, %s", database_name, schema_name, table_name);
-  return K2PgStatus::NotSupported;
+    elog(DEBUG5, "PgGateAPI: PgGate_NewCreateTable %s, %s, %s", database_name, schema_name, table_name);
+    std::vector<k2pg::ColumnSchema> k2pgcols;
+    std::vector<k2pg::ColumnId> colIds;
+    int num_key_columns = 0;
+    const std::string sname = schema_name; // get std::string to make comparison easier
+    const bool is_pg_catalog_table = (sname == "pg_catalog") || (sname == "information_schema");
+    const bool is_shared_table = false; // TODO: Should be passed in by parameter?
+
+    // Add internal primary key column to a Postgres table without a user-specified primary key.
+    if (add_primary_key) {
+        // For regular user table, k2pgrowid should be a hash key because k2pgrowid is a random uuid.
+        bool is_hash = !is_pg_catalog_table;
```

Comment:
All of our keys are range keys for how we are implementing chogori-opengauss. If it is not too much of a refactor we should remove the is_hash concept completely.

Reply:
Removed. Will go though rest of the code to remove it if present.